File: aladdinworksfivefiftyfive_ui/src/services/PowerUsageReportService.js
import http from "../http-common"; 

class PowerUsageReportService {
  getAllPowerUsageReports(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/powerUsageReport/powerUsageReports`, searchDTO);
  }

  get(powerUsageReportId) {
    return this.getRequest(`/powerUsageReport/${powerUsageReportId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/powerUsageReport?field=${matchData}`, null);
  }

  addPowerUsageReport(data) {
    return http.post("/powerUsageReport/addPowerUsageReport", data);
  }

  update(data) {
  	return http.post("/powerUsageReport/updatePowerUsageReport", data);
  }
  
  uploadImage(data,powerUsageReportId) {
  	return http.postForm("/powerUsageReport/uploadImage/"+powerUsageReportId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PowerUsageReportService();

File: src/main/java/com/aladdinworksfivefiftyfive/service/PowerUsageReportService.java
package com.aladdinworksfivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworksfivefiftyfive.domain.PowerUsageReport;
import com.aladdinworksfivefiftyfive.dto.PowerUsageReportDTO;
import com.aladdinworksfivefiftyfive.dto.PowerUsageReportSearchDTO;
import com.aladdinworksfivefiftyfive.dto.PowerUsageReportPageDTO;
import com.aladdinworksfivefiftyfive.dto.PowerUsageReportConvertCriteriaDTO;
import com.aladdinworksfivefiftyfive.service.GenericService;
import com.aladdinworksfivefiftyfive.dto.common.RequestDTO;
import com.aladdinworksfivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PowerUsageReportService extends GenericService<PowerUsageReport, Integer> {

	List<PowerUsageReport> findAll();

	ResultDTO addPowerUsageReport(PowerUsageReportDTO powerUsageReportDTO, RequestDTO requestDTO);

	ResultDTO updatePowerUsageReport(PowerUsageReportDTO powerUsageReportDTO, RequestDTO requestDTO);

    Page<PowerUsageReport> getAllPowerUsageReports(Pageable pageable);

    Page<PowerUsageReport> getAllPowerUsageReports(Specification<PowerUsageReport> spec, Pageable pageable);

	ResponseEntity<PowerUsageReportPageDTO> getPowerUsageReports(PowerUsageReportSearchDTO powerUsageReportSearchDTO);
	
	List<PowerUsageReportDTO> convertPowerUsageReportsToPowerUsageReportDTOs(List<PowerUsageReport> powerUsageReports, PowerUsageReportConvertCriteriaDTO convertCriteria);

	PowerUsageReportDTO getPowerUsageReportDTOById(Integer powerUsageReportId);







}





